### ISSUE TITLE
Silent coercion from logits to probabilities

### ISSUE THREAD

**Issue by beyarkay** [keras-team/tf-keras#47]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): M1 Mac
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): `unknown 2.10.0`
- Python version: `Python 3.10.5`
- Bazel version (if compiling from source): N/A
- GPU model and memory: M1 Mac
- Exact command to reproduce: N/A

**Describe the problem**.

The code for cross entropy will silently coerce the `output` vector so that it will sum to one. It is very easy to forget about the `from_logits` parameter (which is false by default), and mistakenly pass in logits to cross entropy which don't sum to one but which get silently scaled to sum to one regardless.

The code snippet in question: https://github.com/keras-team/keras/blob/master/keras/backend.py#L5589
```py
def categorical_crossentropy(target, output, from_logits=False, axis=-1):
    ...
    # Adjust the predictions so that the probability of
    # each class for every sample adds up to 1
    # This is needed to ensure that the cross entropy is
    # computed correctly.
    output = output / tf.reduce_sum(output, axis, True)
```

This sort of error is easy to catch (by just checking that the output isn't outrageously far from summing to 1) and would solve a lot of common mistakes with keras beginners.

**Describe the current behavior**.

If I mistakenly pass logits to cross entropy loss and forget about the `from_logits` parameter, I will get no error but will get incorrect results

**Describe the expected behavior**.

If I pass something to cross entropy loss that's *obviously* not probabilities, it's probably a user error and the command should fail.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): Add an assertion that fails if the `output` sums to a value that isn't within a tolerance to 1.

**Standalone code to reproduce the issue**.
```py
import tensorflow as tf
import numpy as np
y_true = [[0, 1, 0], [0, 0, 1]]
y_logits = [[0.15, 0.05, 0], [0.01, 0.08, 0.01]]
# Assert that the `y_logits` don't sum to 1 (so they're not valid probabilities)
assert not np.any(np.array(y_logits).sum(axis=1) == 1)
# But we can still pass them in to categorical_crossentropy regardless ):
# no error is omitted
loss = tf.keras.losses.categorical_crossentropy(y_true, y_logits)
# The loss is still calculated fine, even though it's meaningless
assert np.isclose(
    loss.numpy(),
    np.array([1.3862944, 2.302585 ])
).all()
```

Thanks!
